cmd/juju/caas/aks.go

```diff
@@ -0,0 +1,287 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caas
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/juju/cmd"
+	"github.com/juju/collections/set"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/caas/kubernetes/clientconfig"
+	"github.com/juju/juju/cmd/juju/interact"
+)
+
+type aks struct {
+	CommandRunner
+	azExecName string
+}
+
+func newAKSCluster() k8sCluster {
+	return &aks{CommandRunner: &defaultRunner{}}
+}
+
+func (a *aks) cloud() string {
+	return "aks"
+}
+
+func (a *aks) ensureExecutable() error {
+	if a.azExecName != "" {
+		return nil
+	}
+
+	cmd := []string{"which", "az"}
+	err := collapseRunError(runCommand(a, cmd, ""))
+	if err != nil {
```

Comment:
Doing the simple == nil case first reads better
```
if err == nil {
    a.azExecName = "az"
} else {
}
```

Reply:
very true!